### PR TITLE
feat(apigateway): image sourcing support

### PR DESCRIPTION
**Pull request by roleyfoley**
## Intent of Change

- New feature (non-breaking change which adds functionality)

## Description

Adds support for specifying alternate image sources for apigateway

Also includes a none option which can be used in modules which specify the apidefintion and inject it as part of the blueprint processing

## Motivation and Context

Allows for external sources to be used for API Gateway definitions which simplifies the build process and also removes unnecessary pregeneraiton scripts when they aren't required

## How Has This Been Tested?

Tested locally

## Followup Actions

- [x] None

### CODE REVIEW

File: providers/shared/components/apigateway/id.ftl [hunk 1]
```diff
@@ -244,6 +244,37 @@ object.
             {
                 "Names" : "Tracing",
                 "Children" : tracingChildConfiguration
+            },
+            {
+                "Names" : "Image",
+                "Description" : "Control the source of the image that is used for the function",
```
Comment (RossMurr4y) [inline providers/shared/components/apigateway/id.ftl:250]: This description was applicable to `lambda` implementation of these attributes, but could use updated wording for the apigateway

File: providers/shared/components/apigateway/id.ftl [hunk 2]
```diff
@@ -244,6 +244,37 @@ object.
...
+                        "Values" : [ "registry", "url", "none" ],
+                        "Default" : "registry"
+                    },
+                    {
+                        "Names" : "Source:url",
+                        "Description" : "Url Source specific Configuration",
+                        "Children" : [
+                            {
+                                "Names" : "Url",
+                                "Description" : "The Url to the lambda zip file",
```
Comment (RossMurr4y) [inline providers/shared/components/apigateway/id.ftl:266]: here too

**Review by RossMurr4y — APPROVED** [review]
Ready to merge once you're happy with the description.

File: providers/shared/components/apigateway/id.ftl (hunk 2 above)
Comment (roleyfoley) [reply providers/shared/components/apigateway/id.ftl]: Fixed

**Pull request merged**